src/Nethermind.Crypto.SecP256r1/BoringSSL.cs

```diff
@@ -0,0 +1,92 @@
+// SPDX-FileCopyrightText: 2025 Demerzel Solutions Limited
+// SPDX-License-Identifier: LGPL-3.0-only
+
+using System.Reflection;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using System.Runtime.Loader;
+
+namespace Nethermind.Crypto;
+
+public static partial class BoringSSL
+{
+    private const string LibraryName = "secp256r1";
+    static BoringSSL() => SetLibraryFallbackResolver();
+
+    private static void SetLibraryFallbackResolver()
+    {
+        Assembly assembly = typeof(Secp256r1).Assembly;
+
+        AssemblyLoadContext.GetLoadContext(assembly)!.ResolvingUnmanagedDll += (context, name) =>
+        {
+            if (context != assembly || !LibraryName.Equals(name, StringComparison.Ordinal))
+                return nint.Zero;
+
+            string platform;
+
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
```

Comment:
I'm not sure if we have a cross-platform suite test, but if we don't, it would be a great opportunity to create one and then have a setup that would run it on matrix of all supported OSes.

Reply:
Massive! Was not aware of it!